src/snowflake/snowpark/_internal/xml_reader.py

```diff
@@ -233,9 +229,6 @@ def find_next_opening_tag_pos(
         # Update the overlap from the end of the combined data.
         overlap = data[-overlap_size:] if len(data) >= overlap_size else data
 
-        # Otherwise, rewind by the length of the overlap so that a tag spanning the boundary isn't missed.
-        file_obj.seek(-len(overlap), 1)
-
```

Comment:
line 230 is already reading overlapped data into `overlap`. we don't need to seek back. The `absolute_pos` calculation on line 222 also accounts for this.